plugin/src/main/kotlin/org/gradle/dependencygraph/extractor/DependencyExtractor.kt

```diff
@@ -100,14 +101,16 @@ abstract class DependencyExtractor :
         val details: D? = buildOperation.details.let {
             if (it is D) it else null
         }
-        val result: R? = finishEvent.result.let {
-            if (it is R) it else null
+        if (details == null) {
+            return  // Ignore other build operation types
         }
-        if (details == null && result == null) {
-            return
-        } else if (details == null || result == null) {
```

Comment:
I'm curious what causes these types to not align such that one is null but not the other.

I definitely think this is the right fix, but I'm confused why this exception was being thrown

Reply:
See #103 for more background